### ISSUE TITLE
Website crashes when clicking reauthenticate on the settings page

### ISSUE THREAD

**Issue by CarolineGeven** [Yoast/google-analytics-for-wordpress#262]
When clicking the link 'reauthenticate' the website crashes. It requires the techs at the hosting service to kill services and reboot services.

One of the sites was offline for an hour after someone clicked on the "reauthenticate on the settings page" link. This is documented with Pingdom ( http://stats.pingdom.com/rg923mlmjsvr/860545/2014/12 ). Other sites experienced shorter downtime periods -- because of the plugin (e.g., http://stats.pingdom.com/rg923mlmjsvr/860533/2014/12 and http://stats.pingdom.com/rg923mlmjsvr/1198090/2014/12 and http://stats.pingdom.com/rg923mlmjsvr/1126232/2014/12 and so on). Downtime is also documented by the hosting company.

**The hosting company WP Engine reports the following:**

"I was looking into this some more for you, and it looks like that plugin is creating a row in the options table, set to autoload=yes, that is rather large (showing length in bytes):

![](https://cldup.com/ncr6PsK8tc-3000x3000.png)

We can see from my screenshot above that the larger autoloaded data is almost exclusively coming from the `yst_ga_top_pageviews`. So it would seem that clicking the link you are describing is causing this option row to grow, and then holding up memcached with the object being too large, and then clogging up Apache as the processes just wait around to try and load the option again and again until they time out."

**Comment by boogah** [comment]
Jason from WP Engine here. If there's any information or assistance I can provide, let me know!

**Comment by boogah** [comment]
I know you folks are working on tons of stuff, but I wanted to make sure that this doesn't fall through the cracks. If there's anything I (or we) can do to help, please let me know!

**Comment by CarolineGeven** [comment]
Hi Cosper, we've created a beta and sent this to the customer. We've put a limit on the maximum records we're fetching from Google Analytics. We've also added the option to switch off dashboards  and have created a filter to lower the maximum of records to fetch from Google Analytics 
(the filter: `yst-ga-filter-api-limit`). 

We're awaiting reply from the customer if this solves the issue.

Right now Google Analytics has been released with the current fix, so this issue could be tested with the free version of our plugin.
